flow/activities/snapshot_activity.go

```diff
@@ -104,3 +106,54 @@ func (a *SnapshotActivity) SetupReplication(
 		SnapshotName: slotInfo.SnapshotName,
 	}, nil
 }
+
+func (a *SnapshotActivity) MaintainTx(ctx context.Context, sessionID string, peer *protos.Peer) error {
+	conn, err := connectors.GetCDCPullConnector(ctx, peer)
+	if err != nil {
+		return err
+	}
+	defer connectors.CloseConnector(ctx, conn)
+
+	snapshotName, tx, err := conn.ExportSnapshot(ctx)
+	if err != nil {
+		return err
+	}
+
+	sss := SlotSnapshotSignal{snapshotName: snapshotName}
+	a.SnapshotConnectionsMutex.Lock()
+	a.SnapshotConnections[sessionID] = sss
+	a.SnapshotConnectionsMutex.Unlock()
+
+	for {
+		activity.RecordHeartbeat(ctx, "maintaining export snapshot transaction")
```

Comment:
can we log something here as well, and make the message for both heartbeat and log something like.

`maintaining export snapshot transaction since X mins`.